lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -651,12 +652,26 @@ class InheritedProtocolCollector {
     // No recursion here because extensions are never nested.
   }
 
+  /// Returns true if the conformance of \p nominal to \p proto is declared in
+  /// module \p M.
+  static bool conformanceDeclaredInModule(ModuleDecl *M,
+                                          const NominalTypeDecl *nominal,
+                                          ProtocolDecl *proto) {
+    SmallVector<ProtocolConformance *, 4> conformances;
+    nominal->lookupConformance(M, proto, conformances);
```

Comment:
@DougGregor I wasn't super happy about using this query because it has to build the whole lookup table, but maybe we can assume that's happened by the time we're done compiling a module anyway?

Reply:
There might be a way to do what you want (printing conformed-to protocols, either directly or via inheritance, for public protocols only) by walking the ConformanceLookupTable directly.

Reply:
Hm, right. This can still result in a conformance being printed twice.

Reply:
Yeah, given what's there nothing jumps out at me as being better than this.